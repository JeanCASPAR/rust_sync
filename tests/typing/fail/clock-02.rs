use rustre::sync;

sync! {
    #![pass(1)]

    node oui() = ()
    where
        a : float = 3.0 when a;
}

fn main() {}
