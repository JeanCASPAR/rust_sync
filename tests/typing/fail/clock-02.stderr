error: type error: the clock `a` should have type `bool`, found `float`
 --> tests/typing/fail/clock-02.rs:8:30
  |
8 |         a : float = 3.0 when a;
  |                              ^
  |
help: `a` was declared here with type `float`
 --> tests/typing/fail/clock-02.rs:8:9
  |
8 |         a : float = 3.0 when a;
  |         ^
